code/go/pkg/validator/validator_test.go

```diff
@@ -119,6 +119,15 @@ func TestValidateFile(t *testing.T) {
 				fmt.Sprintf("field ilm_policy: ILM policy \"logs-bad_custom_ilm_policy.test-notexists\" not found in package, expected definition in \"%sbad_custom_ilm_policy/data_stream/test/elasticsearch/ilm/notexists.json\"", osTestBasePath),
 			},
 		},
+		"bad_select": {
+			"data_stream/foo_stream/manifest.yml",
+			[]string{
+				"field streams.0.vars.1: Must validate \"then\" as \"if\" was valid",
+				"field streams.0.vars.1: options is required",
+				"field streams.0.vars.1: Must validate \"then\" as \"if\" was valid",
+				"field streams.0.vars.2.options: Invalid type. Expected: array, given: null",
+			},
+		},
```

Comment:
It is not ideal that each failed validation gives two error messages:
1. `field streams.0.vars.1: Must validate "then" as "if" was valid`
2. And the actual error message.

Also I wish the error message for when `options` is an empty array was better than `Expected: array, given: null`.

Any thoughts?

Reply:
> Also I wish the error message for when options is an empty array was better than Expected: array, given: null.
weird, maybe the gojsonschema lib transforms the empty array to null ? 


About the error messages (two messages per each error), as they are the ones reported by the jsonschema validation, I think it cannot be done nothing here.

Reply:
Agreed. The new error message (when `options` are passed when `type` is `text`) is even worse 🙂